### ISSUE TITLE
Doesn't play MKV

### ISSUE THREAD

**Issue by HOSAINT** [anilbeesetti/nextplayer#339]
![Screenshot_2023-06-05-16-43-07-322_dev anilbeesetti nextplayer](https://github.com/anilbeesetti/nextplayer/assets/62555951/dfd1692a-a5a3-4eaa-ab81-29dafa54ea71)

The error is on Xiaomi Redmi 9. I don't know if it's phone related or not. I installed ver0.2.0 universal and arm7
The video player doesn't show thumbnails for videos it does not play.

**Comment by Pentaphon** [comment]
Your phone doesn't support hardware decode of 10 bit AVC files.

**Comment by MrDark-code** [comment]
Say, if there is no hardware decoder support, shouldn't it automatically switch to a software decoder in such a case ?

**Comment by anilbeesetti** [comment]
App doesn't have software video decoders that support 10bit video yet. They will be added soon

**Comment by anilbeesetti** [comment]
@HOSAINT is the video playing on new release?

**Comment by MrDark-code** [comment]
@anilbeesetti 
On Redmi note 9 (Android 10 MIUI 11.5.0) there is a problem with this file from English. sound track 
[test hevc 10 bit SW](https://link.storjshare.io/s/jxydsglq3pc6zkzw4ztchfsivqrq/vnhghhv/Keijo.S01E03.mkv?wrap=0)

**Comment by Pentaphon** [comment]
> On Redmi note 9 (Android 10 MIUI 11.5.0) there is a problem with this file from English. sound track
> [test hevc 10 bit SW](https://link.storjshare.io/s/jxydsglq3pc6zkzw4ztchfsivqrq/vnhghhv/Keijo.S01E03.mkv?wrap=0)

On 0.5.0 I get an unexpected runtime error when I switch to the 2nd audio track.

**Comment by MrDark-code** [comment]
@Pentaphon 
Yes, this is the problem and when i switched to the English sound track, it seems that only the background of the voice acting is playing there without a voice.
On the Japanese track, the sound goes well there.🤔

**Comment by MrDark-code** [comment]
@anilbeesetti 
the visibility of the player needs to be slightly improved, for example, in the soul browser in the header, the link the player following him is not in the list

**Comment by anilbeesetti** [comment]
> @anilbeesetti the visibility of the player needs to be slightly improved, for example, in the soul browser in the header, the link the player following him is not in the list

I don't understand what you mean, could you elaborate

**Comment by MrDark-code** [comment]
@anilbeesetti 
Share URL button
no Next player
![Screenshot_2023-07-25-20-48-47-344_android](https://github.com/anilbeesetti/nextplayer/assets/137885849/eca6ff33-7a55-49de-b246-256fb3e054a2)

**Comment by MrDark-code** [comment]
@anilbeesetti  version 0.8.1 and 0.6.0

[video screen recorder ](https://link.storjshare.io/s/jxddfz2a3xfuswefhddsk7qqwuzq/vnhghhv/Screenrecorder-2023-09-15-09-34-39-676.mp4?wrap=0)

**Comment by MrDark-code** [comment]
@anilbeesetti  v. 0.6.0
[Video 10 bit problem with audio 5.1 decoder SW](https://link.storjshare.io/s/jucur4limecxpasibkwtz4563wcq/vnhghhv/Screenrecorder-2023-09-15-10-47-31-17.mp4?wrap=0)

**Comment by anilbeesetti** [comment]
> @anilbeesetti version 0.8.1 and 0.6.0
> 
> [video screen recorder ](https://link.storjshare.io/s/jxddfz2a3xfuswefhddsk7qqwuzq/vnhghhv/Screenrecorder-2023-09-15-09-34-39-676.mp4?wrap=0)

@MrDark-code Is app crashing when playing all videos or some specific videos. If specific video then it would be helpful if you share that video

**Comment by MrDark-code** [comment]
@anilbeesetti all video yuv420p and yuv420p10bit and codecs 

As for the sound in the Mx player, it works fine because it is converted to stereo

**Comment by anilbeesetti** [comment]
@MrDark-code could you share your device info and test video to check

**Comment by MrDark-code** [comment]
@anilbeesetti  Redmi note 9 MIUI Global 11.5.0.
Ok go

**Comment by MrDark-code** [comment]
you also need to add media info to ffmpeg

**Comment by anilbeesetti** [comment]
Will add mediainfo, it will take time. And what's the android version?

**Comment by MrDark-code** [comment]
@anilbeesetti  10

**Comment by MrDark-code** [comment]
@anilbeesetti [V. 0.9.0 all codecs](https://link.storjshare.io/s/jva5kkzayll6mxeyxnqmclkkb2mq/vnhghhv/Screenrecorder-2023-09-30-20-51-06-264.mp4?wrap=0)

**Comment by ngocanhtve** [comment]
> App doesn't have software video decoders that support 10bit video yet. They will be added soon

I also want to report an issue related to video 10bit color.

**Video**: AV1, 10 bit color, MKV format.
**Audio**: 5.1 Opus @ 192 kbps.
**CPU**: Snapdragon 855 (Sony Xperia 1).
**NextPlayer**: Latest.
**Android**: 13.
**Issue**: Image movement is slow when playing video at 1.5x speed. It is like kinda were played at 0.5x.

**Comment by anilbeesetti** [comment]
I will try to add software av1 decoder soon.

**Comment by MrDark-code** [comment]
@anilbeesetti hello, the new version is working, but mediaifo only works for local files, it’s not conveniently implemented, I need it for streaming files, as for example it’s implemented in this application
[ffplayer](https://www.absonux.com/ffplayer)
![Screenshot_2023-11-10-10-46-33-250_com.absonux.ffplayer.jpg](https://github.com/anilbeesetti/nextplayer/assets/137885849/dd4bdae3-565d-46c7-a3f8-0262672dcaed)

**Comment by MrDark-code** [comment]
@anilbeesetti hello, the new version is working, but mediainfo only works for local files, it’s not conveniently implemented, me need it for streaming files, as for example it’s implemented in this application
[ffplayer](https://www.absonux.com/ffplayer)
![Screenshot_2023-11-10-10-46-33-250_com.absonux.ffplayer.jpg](https://github.com/anilbeesetti/nextplayer/assets/137885849/dd4bdae3-565d-46c7-a3f8-0262672dcaed)

Add
![sketch-1699589636636.png](https://github.com/anilbeesetti/nextplayer/assets/137885849/5e2606b9-fdfd-48c7-90c5-0f0476eff562)

**Comment by anilbeesetti** [comment]
> @anilbeesetti hello, the new version is working, but mediainfo only works for local files, it’s not conveniently implemented, me need it for streaming files, as for example it’s implemented in this application [ffplayer](https://www.absonux.com/ffplayer) ![Screenshot_2023-11-10-10-46-33-250_com.absonux.ffplayer.jpg](https://user-images.githubusercontent.com/137885849/281946019-dd4bdae3-565d-46c7-a3f8-0262672dcaed.jpg)
> 
> Add ![sketch-1699589636636.png](https://user-images.githubusercontent.com/137885849/281953611-5e2606b9-fdfd-48c7-90c5-0f0476eff562.png)

Will add in the next release.

**Comment by bmsrrb** [comment]
It seems the functionality is not there yet @anilbeesetti 

> Source error

on a MKV video (video/x-matroska)

Any updates?

**Comment by anilbeesetti** [comment]
@bmsrrb, Did you test with new version https://github.com/anilbeesetti/nextplayer/releases/tag/v0.18.0. You can try the different decoder with this release. try HW, SW+, SW. If still not working can share the video file along with your device details and also a screen recording if possible

**Comment by bmsrrb** [comment]
> [@bmsrrb](https://github.com/bmsrrb), Did you test with new version https://github.com/anilbeesetti/nextplayer/releases/tag/v0.18.0. You can try the different decoder with this release. try HW, SW+, SW. If still not working can share the video file along with your device details and also a screen recording if possible

Just tried. Not working.

The video is from this torrent: https://x1337x.cc/torrent/251634/Steve-Jobs-Pirates-of-Silicon-Valley-1999-DVDRip-REQU3ST/

Screenshot: right after asked to play the video

<img width="576" height="1289" alt="Screenshot of Next Player showing an Error on an MKV file " src="https://github.com/user-attachments/assets/7c5f6b5e-0b08-4619-a986-852f3d961c85" />

Note: It plays on MX player 1.93.4
